### ISSUE TITLE
cabal install not working?

### ISSUE THREAD

**Issue by trotha01** [elm-lang/elm-lang.org#346]
maybe it's just me, but cabal install is having some dependency issues:
```
$ cabal install --only-dependencies
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: Diff-0.3.1/installed-236... (user goal)
trying: QuickCheck-2.8/installed-1b3... (user goal)
trying: elm-website-0.1.0.2 (user goal)
trying: elm-package-0.5 (dependency of elm-website-0.1.0.2)
next goal: optparse-applicative (dependency of elm-package-0.5)
rejecting: optparse-applicative-0.11.0.2, 0.11.0.1, 0.11.0 (conflict:
elm-package => optparse-applicative>=0.8.1 && <0.11)
rejecting: optparse-applicative-0.10.0, 0.9.1.1, 0.9.1, 0.9.0 (conflict: Diff
=> base==4.8.0.0/installed-901..., optparse-applicative => base>=4 && <4.8)
rejecting: optparse-applicative-0.8.1 (conflict: QuickCheck =>
transformers==0.4.2.0/installed-c1a..., optparse-applicative =>
transformers>=0.2 && <0.4)
rejecting: optparse-applicative-0.8.0.1, 0.8.0, 0.7.0.2, 0.7.0.1, 0.7.0,
0.6.0, 0.5.2.1, 0.5.2, 0.5.1, 0.5.0, 0.4.3, 0.4.2, 0.4.1, 0.4.0, 0.3.2, 0.3.1,
0.3.0, 0.2.0, 0.1.1, 0.1.0, 0.0.1 (conflict: elm-package =>
optparse-applicative>=0.8.1 && <0.11)
Backjump limit reached (change with --max-backjumps).

Note: when using a sandbox, all packages are required to have consistent
dependencies. Try reinstalling/unregistering the offending packages or
recreating the sandbox.
```

**Comment by trotha01** [comment]
hmm, I re-did the sandbox with `cabal sandbox delete`, `cabal sandbox init`, `cabal configure` and install seemed to give a different dependency error:
```
$ cabal install --only-dependencies
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: elm-website-0.1.0.2 (user goal)
trying: snap-core-0.9.7.2 (dependency of elm-website-0.1.0.2)
trying: vector-0.11.0.0 (dependency of snap-core-0.9.7.2)
next goal: elm-package (dependency of elm-website-0.1.0.2)
rejecting: elm-package-0.5, 0.4, 0.2.2, 0.2.1, 0.2 (conflict:
vector==0.11.0.0, elm-package => vector>=0.10 && <0.11)
Backjump limit reached (change with --max-backjumps).

Note: when using a sandbox, all packages are required to have consistent
dependencies. Try reinstalling/unregistering the offending packages or
recreating the sandbox.
```

**Comment by evancz** [comment]
I have `snap-core-0.9.7.0` and `vector-0.10.12.3`, so I wonder if the latest patch version changed the dependency on `vector`?

If you run `cabal install --constraint="snap-core==0.9.7.0"` does it have better results? Otherwise I don't have a short term fix for this besides blowing up my current development environment to fix this up. Perhaps the rule for elm-lang.org should be to use exact dependencies in the .cabal file?

**Comment by ksmk** [comment]
Hi, same problem here 

cabal: Could not resolve dependencies:
trying: elm-website-0.1.0.2 (user goal)
trying: snap-core-0.9.7.2 (dependency of elm-website-0.1.0.2)
trying: vector-0.11.0.0 (dependency of snap-core-0.9.7.2)
next goal: elm-package (dependency of elm-website-0.1.0.2)
rejecting: elm-package-0.5, 0.4, 0.2.2, 0.2.1, 0.2 (conflict:
vector==0.11.0.0, elm-package => vector>=0.10 && <0.11)
Dependency tree exhaustively searched.

( on platform with cabal 1.22 and ghc 7.10.2 )

**Comment by evancz** [comment]
It looks like packages have changed in hackage in a way that doesn't work anymore.

I'd need to blow up my whole development environment to reproduce this, so I don't know if I'll get a chance to do it before you. The way to resolve this is to mess with the version bounds in elm-website.cabal until things start working. If you are able to figure out how to restrict things to make it solve, please send a PR along!

**Comment by ksmk** [comment]
Looks like all comes down to resolving dependencies / finding out how to compile dependencies in elm-package. I'll try! :)

**Comment by jvoigtlaender** [comment]
Suggestion: Try the [Stackage/LTS Haskell way](https://www.stackage.org/). By putting [this file](https://www.stackage.org/lts-3.0/cabal.config) into the project root, then calling `cabal update`, you start from a collection of packages that are not only guaranteed to fit together in terms of declared version numbers, but also are guaranteed to actually *compile* all together (thus addressing "Cabal hell" arising from people declaring version numbers that do not strictly adher to semantic versioning). It covers only a subset of Hackage packages, but a big one. It might be enough to compile Elm. If not, there's a process to get needed packages into this automatically-vetted-and-guaranteed-to-compile-together set-of-packages.

**Comment by jvoigtlaender** [comment]
I have just verified that "Long Term Support Haskell 3.1" is compatible with the bounds on required packages here.

So: The above problems should disappear if after the
```bash
cabal sandbox init --sandbox ..
```
command in the install instructions you do a
```bash
curl https://www.stackage.org/lts-3.1/cabal.config > cabal.config
```
and a
```bash
cabal update
```
This will automatically pick compatible versions of `vector` and `snap-core` etc.

**Comment by ksmk** [comment]
Thx for the tip but I still get
"Resolving dependencies...
cabal: Could not resolve dependencies:
trying: elm-website-0.1.0.2 (user goal)
trying: elm-package-0.5 (dependency of elm-website-0.1.0.2)
next goal: optparse-applicative (dependency of elm-package-0.5)
rejecting: optparse-applicative-0.11.0.2 (conflict: elm-package =>
optparse-applicative>=0.8.1 && <0.11)
rejecting: optparse-applicative-0.11.0.1, 0.11.0, 0.10.0, 0.9.1.1, 0.9.1,
0.9.0, 0.8.1, 0.8.0.1, 0.8.0, 0.7.0.2, 0.7.0.1, 0.7.0, 0.6.0, 0.5.2.1, 0.5.2,
0.5.1, 0.5.0, 0.4.3, 0.4.2, 0.4.1, 0.4.0, 0.3.2, 0.3.1, 0.3.0, 0.2.0, 0.1.1,
0.1.0, 0.0.1 (global constraint requires ==0.11.0.2)
Dependency tree exhaustively searched.

Note: when using a sandbox, all packages are required to have consistent
dependencies. Try reinstalling/unregistering the offending packages or
recreating the sandbox."

**Comment by ksmk** [comment]
been hammering but I'm stuck upon the language-glsl-* install: 

Resolving dependencies...
cabal: Could not resolve dependencies:
trying: elm-website-0.1.0.2 (user goal)
trying: elm-compiler-0.15 (dependency of elm-website-0.1.0.2)
next goal: language-glsl (dependency of elm-compiler-0.15)
rejecting: language-glsl-0.2.0 (conflict: elm-compiler => language-glsl>=0.0.2
&& <0.2)
rejecting: language-glsl-0.1.1, 0.1.0, 0.0.2, 0.0.1, 0.0.0 (global constraint
requires ==0.2.0)
Dependency tree exhaustively searched.

**Comment by ksmk** [comment]
'stack init --solver ' 'resolver:nightly-2015-08-17'
but again conflicts:

trying: elm-website-0.1.0.2 (user goal)
trying: mtl-2.2.1 (dependency of elm-website-0.1.0.2)
trying: elm-package-0.5 (dependency of elm-website-0.1.0.2)
next goal: optparse-applicative (dependency of elm-package-0.5)
rejecting: optparse-applicative-0.11.0.2, 0.11.0.1, 0.11.0 (conflict:
elm-package => optparse-applicative>=0.8.1 && <0.11)
rejecting: optparse-applicative-0.10.0, 0.9.1.1, 0.9.1, 0.9.0 (conflict:
base==4.8.1.0/installed-4f7..., optparse-applicative => base>=4 && <4.8)
trying: optparse-applicative-0.8.1
next goal: transformers (dependency of mtl-2.2.1)
rejecting: transformers-0.4.2.0/installed-21d..., 0.4.3.0, 0.4.2.0, 0.4.1.0
(conflict: optparse-applicative => transformers>=0.2 && <0.4)
rejecting: transformers-0.3.0.0, 0.2.2.1, 0.2.2.0, 0.2.1.0, 0.2.0.0, 0.1.4.0,
0.1.3.0, 0.1.1.0, 0.1.0.1, 0.1.0.0, 0.0.1.0, 0.0.0.0 (conflict: mtl =>
transformers==0.4.*)
rejecting: transformers-0.4.0.0 (conflict: optparse-applicative =>
transformers>=0.2 && <0.4)
Dependency tree exhaustively searched.

**Comment by jvoigtlaender** [comment]
Well, are you actually following the install instructions from the README of this repository? Specifically, have you run [this script](https://github.com/elm-lang/elm-platform/blob/master/installers/BuildFromSource.hs) before the three install attempts posted this night? (I see only one of them left above.)

I conjecture you haven't, because in your logs I see things like `trying: elm-compiler-0.15` and `trying: elm-package-0.5`. But these are deprecated (see on Hackage)! If you had run the linked script as instructed, you should have recent versions (not-on-Hackage) of `elm-compiler` and `elm-package` already in your relevant sandbox. Then no `trying` with older versions should appear in the logs.

(Note that my suggestion above was only about *adding* something to the install instructions from the README. I didn't say that you can in exchange leave out some other step, like running the linked script. :smile:)

**Comment by jvoigtlaender** [comment]
Maybe you did run the linked script but are running into [the problem discussed here](https://github.com/elm-lang/elm-platform/issues/41#issuecomment-132043962). Then the trick from there might work, and any solution found there might also work. It is unrelated to whether or not "LTS Haskell" is used.

**Comment by ksmk** [comment]
jvoigtlaender, thx for being so quick and supportive!
I've put elm-thing aside for a while but I'm going to figure it out now for sure.

**Comment by trotha01** [comment]
So I re-ran
``` curl https://www.stackage.org/lts-3.1/cabal.config > cabal.config ```

and
``` cabal update ```

but I'm still getting errors:
```
$ cabal install --only-dependencies
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: aeson-0.8.0.2/installed-b70... (user goal)
next goal: attoparsec (user goal)
rejecting: attoparsec-0.12.1.3/installed-575..., 0.13.0.1, 0.13.0.0 (global
constraint requires ==0.12.1.6)
rejecting: attoparsec-0.12.1.6 (conflict: aeson =>
attoparsec==0.12.1.3/installed-575...)
rejecting: attoparsec-0.12.1.5, 0.12.1.4, 0.12.1.3, 0.12.1.2, 0.12.1.1,
0.12.1.0, 0.12.0.0, 0.11.3.4, 0.11.3.3, 0.11.3.2, 0.11.3.1, 0.11.3.0,
0.11.2.1, 0.11.1.0, 0.10.4.0, 0.10.3.0, 0.10.2.0, 0.10.1.1, 0.10.1.0,
0.10.0.3, 0.10.0.2, 0.10.0.1, 0.10.0.0, 0.9.1.2, 0.9.1.1, 0.9.0.0, 0.8.6.1,
0.8.6.0, 0.8.5.3, 0.8.5.2, 0.8.5.1, 0.8.5.0, 0.8.4.0, 0.8.3.0, 0.8.2.0,
0.8.1.1, 0.8.1.0, 0.8.0.2, 0.8.0.1, 0.8.0.0, 0.7.2, 0.7.1, 0.6, 0.5.1, 0.5,
0.4 (global constraint requires ==0.12.1.6)
Dependency tree exhaustively searched.

Note: when using a sandbox, all packages are required to have consistent
dependencies. Try reinstalling/unregistering the offending packages or
recreating the sandbox.
```

**Comment by jvoigtlaender** [comment]
Ah, things have changed since last month when the above comments were made. In particular, the [build script](https://github.com/elm-lang/elm-platform/blob/master/installers/BuildFromSource.hs) has changed since then.

Have you run the current version, or one that you downloaded some while ago?

You should use the current version. Also, with that current version, the README in the current repo here is a bit outdated (I'll send a pull request to fix it). In short, one change that definitely needs to go into it is to replace the `cabal sandbox init --sandbox ..` by `cabal sandbox init --sandbox ../.cabal-sandbox`.

Can you try that? While doing that, you might do without the `curl https://www.stackage.org/lts-3.1/cabal.config > cabal.config` stuff. It might not be needed anymore (or may even be counterproductive) with the newer build script. (And besides, by now the current version of `lts-3` would be `lts-3.5`.)

**Comment by trotha01** [comment]
haha, you guys are moving too quick for me.

Okay, so I re-ran
```
$ runhaskell BuildFromSource.hs 0.15.1
```
then
```
cd Elm-Platform/0.15.1
export PATH=/path/to/Elm-Platform/0.15.1/bin:$PATH
git clone https://github.com/elm-lang/elm-lang.org.git
cd elm-lang.org
cabal update
curl https://www.stackage.org/lts-3.1/cabal.config > cabal.config
```

here's the output for the sandboxing:
```
$ cabal sandbox init --sandbox ../.cabal-sandbox
Writing a default package environment file to
/Users/trevorrothaus/Dropbox/Documents/www/elm/Elm-Platform/0.15.1/elm-lang.org/cabal.sandbox.config
Using an existing sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/Elm-Platform/0.15.1/.cabal-sandbox
```

and here's the output for the cabal install:
```
$ cabal install --only-dependencies
Resolving dependencies...
cabal: Could not resolve dependencies:
trying: elm-website-0.1.0.2 (user goal)
trying: elm-package-0.5 (dependency of elm-website-0.1.0.2)
next goal: optparse-applicative (dependency of elm-package-0.5)
rejecting: optparse-applicative-0.11.0.2 (conflict: elm-package =>
optparse-applicative>=0.8.1 && <0.11)
rejecting: optparse-applicative-0.11.0.1, 0.11.0, 0.10.0, 0.9.1.1, 0.9.1,
0.9.0, 0.8.1, 0.8.0.1, 0.8.0, 0.7.0.2, 0.7.0.1, 0.7.0, 0.6.0, 0.5.2.1, 0.5.2,
0.5.1, 0.5.0, 0.4.3, 0.4.2, 0.4.1, 0.4.0, 0.3.2, 0.3.1, 0.3.0, 0.2.0, 0.1.1,
0.1.0, 0.0.1 (global constraint requires ==0.11.0.2)
Dependency tree exhaustively searched.

Note: when using a sandbox, all packages are required to have consistent
dependencies. Try reinstalling/unregistering the offending packages or
recreating the sandbox.
```

let me know if I missed some steps, or if I'm running from the wrong directory, or doing something silly.

**Comment by jvoigtlaender** [comment]
Okay, a few things:
* Are you sure you have been following the latest build instructions from https://github.com/elm-lang/elm-platform? I see you are adding `0.15.1/bin` to your path, while the build instructions say you should have been adding `0.15.1/.cabal-sandbox/bin`. (The instructions in this repo here were still saying to add plain `bin` without `.cabal-sandbox/` yesterday, but that was already fixed last night by merging #374.) In any case, if you do actually have a directory `0.15.1/bin`, this indicates you didn't use the latest version of `BuildFromSource.hs`.
* In my comment here yesterday I said that the https://www.stackage.org/lts stuff might be counterproductive with the new build script. So can you try without the `curl https://www.stackage.org/lts-3.1/cabal.config > cabal.config` line?
* It's suspicious that your `cabal install --only-dependencies` tries to install `elm-package-0.5`. If your preceding `runhaskell BuildFromSource.hs 0.15.1` run was successful, then `elm-package-0.5.1` should already be present. So, let's first make sure that `BuildFromSource.hs` was actually successful. Can you look, after running that script, into `0.15.1/.cabal-sandbox/bin` and tell us what executables you have there? Also, run each of them with with `--help` to see what version they are at.

Of course, all that should be done in a completely fresh directory to start with.

**Comment by jvoigtlaender** [comment]
Additional note: For successfully building `0.15.1`, you may need to incorporate https://github.com/elm-lang/elm-platform/pull/82 into `BuildFromSource.hs` first. (This might or might not have been an issue for you. If `runhaskell BuildFromSource.hs 0.15.1` was already successful, then probably this was not affecting you.)

**Comment by trotha01** [comment]
My bad, I didn't look closely enough at your  https://www.stackage.org/lts comment.

So I destroyed the 0.15.1 directory and re-ran
``` runhaskell BuildFromSource.hs 0.15.1 ```

set the path to include
``` 0.15.1/.cabal-sandbox/bin ```

then
```
git clone https://github.com/elm-lang/elm-lang.org.git
cd elm-lang.org
cabal sandbox init --sandbox ../.cabal-sandbox
```

and I get with cabal install:
```
$ cabal install --only-dependencies
Resolving dependencies...
Notice: installing into a sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox
Downloading utf8-string-1.0.1.1...
Failed to install snap-core-0.9.8.0
Build log ( /Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox/logs/snap-core-0.9.8.0.log ):
cabal:
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox/logs/snap-core-0.9.8.0.log:
does not exist
```

**Comment by jvoigtlaender** [comment]
Okay, so the issue has nothing to do with this repository here. The failure occurs already earlier with installing the other Elm packages. This shifts the question to what happens there with `snap-core-0.9.8.0`. The above log output is not helping. To get more information, I suggest running `BuildFromSource.hs` again, but forcing it to produce more output. You can do that by first removing all the occurrences of `, "-j"` from the file (three places). Hopefully this will give us more information about the failure.

**Comment by jvoigtlaender** [comment]
Minor notice: You actually have to set the path to include `0.15.1/.cabal-sandbox/bin` *before* you run `BuildFromSource.hs`. Mentioned here: https://github.com/elm-lang/elm-platform/blob/master/README.md.

**Comment by trotha01** [comment]
That's a little strange to set it before running BuildFromSource.hs, since the BuildFromSource created that directory for me.

here's the output from running it again:
```
$ runhaskell installers/BuildFromSource.hs 0.15.1
fatal: destination path 'elm-compiler' already exists and is not an empty directory.
fatal: destination path 'elm-package' already exists and is not an empty directory.
fatal: destination path 'elm-make' already exists and is not an empty directory.
fatal: destination path 'elm-reactor' already exists and is not an empty directory.
fatal: destination path 'elm-repl' already exists and is not an empty directory.
Downloading the latest package list from hackage.haskell.org
Writing a default package environment file to
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/cabal.sandbox.config
Using an existing sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox
Some add-source dependencies have been modified. They will be reinstalled...
Resolving dependencies...
Notice: installing into a sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox
Configuring websockets-0.9.5.0...
Failed to install snap-core-0.9.8.0
Build log ( /Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox/logs/snap-core-0.9.8.0.log ):
cabal:
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox/logs/snap-core-0.9.8.0.log:
does not exist
Some add-source dependencies have been modified. They will be reinstalled...
Resolving dependencies...
In order, the following will be installed:
elm-compiler-0.15.1 (reinstall)
elm-package-0.5.1 (reinstall) changes: ansi-wl-pprint-0.6.7.3 added,
optparse-applicative-0.11.0.2 added, pretty-1.1.2.0 added
elm-make-0.2 (new package)
elm-repl-0.4.2 (new package)
Warning: Note that reinstalls are always dangerous. Continuing anyway...
Notice: installing into a sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox
Configuring elm-compiler-0.15.1...
Building elm-compiler-0.15.1...
Installed elm-compiler-0.15.1
Configuring elm-package-0.5.1...
Building elm-package-0.5.1...
Installed elm-package-0.5.1
Configuring elm-make-0.2...
Configuring elm-repl-0.4.2...
Building elm-repl-0.4.2...
Building elm-make-0.2...
Installed elm-repl-0.4.2
Installed elm-make-0.2
Resolving dependencies...
Notice: installing into a sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox
Failed to install snap-core-0.9.8.0
Build log ( /Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox/logs/snap-core-0.9.8.0.log ):
cabal:
/Users/trevorrothaus/Dropbox/Documents/www/elm/elm-platform/Elm-Platform/0.15.1/.cabal-sandbox/logs/snap-core-0.9.8.0Configuring websockets-0.9.5.0...
.log:
does not exist
```

**Comment by jvoigtlaender** [comment]
About the path, yes, it might seem strange, but that doesn't change the fact that it is required. :smile:

Other than that, what I see above is that you didn't run stuff in an empty directory. Right?

**Comment by trotha01** [comment]
What do you mean run in an empty directory?
I'm running it from the directory I cloned the elm-platform repo (https://github.com/elm-lang/elm-platform) into.
Is there someplace specific I should be running it?

**Comment by jvoigtlaender** [comment]
Also, let's try to isolate the problem with `snap-core-0.9.8.0`. Can you do the following?
* Create a new directory.
* In it, `cabal sandbox init`.
* Then, `cabal install snap-core-0.9.8.0 --dry-run` (and post here what the output is of that).
* Finally,  `cabal install snap-core-0.9.8.0`.

**Comment by jvoigtlaender** [comment]
I was referring to the `fatal: destination path 'elm-compiler' already exists and is not an empty directory.` messages. They indicate that you ran `BuildFromSource.hs` in a directory that was not empty.

**Comment by jvoigtlaender** [comment]
And note: https://github.com/elm-lang/elm-platform/blob/master/README.md doesn't tell you to clone that repository. It just advises to download the single file https://github.com/elm-lang/elm-platform/blob/master/installers/BuildFromSource.hs.

**Comment by trotha01** [comment]
yes, I was running it a second time, since you suggested I run it after updating the path. So the directory was not empty.

I actually did only download the single file the first time I ran through this. But looking through the readme again, I didn't see where it specifies to do that.

Looking again now, I should have read the "if you are on windows" comment closer. I'm not on windows, so I saw those first 5 words, and skipped that command

```
$ cabal install snap-core-0.9.8.0 --dry-run
Resolving dependencies...
In order, the following would be installed (use -v for more details):
HUnit-1.3.0.0
bytestring-builder-0.10.6.0.0
bytestring-mmap-0.2.2
extensible-exceptions-0.1.1.4
monads-tf-0.1.0.2
MonadCatchIO-transformers-0.3.1.3
mtl-2.2.1
network-2.6.2.1
old-locale-1.0.0.7
primitive-0.6
random-1.1
regex-base-0.93.2
regex-posix-0.95.2
stm-2.4.4
text-1.2.1.3
blaze-builder-0.4.0.1
enumerator-0.4.20
hashable-1.2.3.3
case-insensitive-1.2.0.4
scientific-0.3.3.8
attoparsec-0.13.0.1
attoparsec-enumerator-0.3.4
unix-compat-0.4.1.4
unordered-containers-0.2.5.1
vector-0.11.0.0
zlib-0.6.1.1
streaming-commons-0.1.13
blaze-builder-enumerator-0.2.1.0
zlib-bindings-0.1.1.5
zlib-enum-0.2.3.1
snap-core-0.9.8.0
```

and
```
 $ cabal install snap-core-0.9.8.0
Resolving dependencies...
Notice: installing into a sandbox located at
/Users/trevorrothaus/Dropbox/Documents/www/elm/test/.cabal-sandbox
Configuring HUnit-1.3.0.0...
Configuring bytestring-builder-0.10.6.0.0...
Configuring bytestring-mmap-0.2.2...
Configuring extensible-exceptions-0.1.1.4...
Building extensible-exceptions-0.1.1.4...
Building bytestring-mmap-0.2.2...
Building HUnit-1.3.0.0...
Building bytestring-builder-0.10.6.0.0...
Installed bytestring-builder-0.10.6.0.0
Configuring monads-tf-0.1.0.2...
Configuring mtl-2.2.1...
Installed extensible-exceptions-0.1.1.4
Building monads-tf-0.1.0.2...
Building mtl-2.2.1...
Configuring network-2.6.2.1...
Installed bytestring-mmap-0.2.2
Configuring old-locale-1.0.0.7...
Installed HUnit-1.3.0.0
Configuring primitive-0.6...
Installed monads-tf-0.1.0.2
Building old-locale-1.0.0.7...
Configuring random-1.1...
Installed mtl-2.2.1
Building primitive-0.6...
Building random-1.1...
Building network-2.6.2.1...
Configuring stm-2.4.4...
Installed old-locale-1.0.0.7
Building stm-2.4.4...
Configuring text-1.2.1.3...
Installed stm-2.4.4
Configuring unix-compat-0.4.1.4...
Installed primitive-0.6
Building text-1.2.1.3...
Building unix-compat-0.4.1.4...
Configuring zlib-0.6.1.1...
Installed unix-compat-0.4.1.4
Installed random-1.1
Configuring MonadCatchIO-transformers-0.3.1.3...
Building zlib-0.6.1.1...
Building MonadCatchIO-transformers-0.3.1.3...
Configuring regex-base-0.93.2...
Installed MonadCatchIO-transformers-0.3.1.3
Building regex-base-0.93.2...
Configuring vector-0.11.0.0...
Installed regex-base-0.93.2
Installed network-2.6.2.1
Configuring regex-posix-0.95.2...
Building vector-0.11.0.0...
Building regex-posix-0.95.2...
Installed zlib-0.6.1.1
Configuring zlib-bindings-0.1.1.5...
Building zlib-bindings-0.1.1.5...
Installed zlib-bindings-0.1.1.5
Installed regex-posix-0.95.2
Installed text-1.2.1.3
Configuring blaze-builder-0.4.0.1...
Configuring hashable-1.2.3.3...
Configuring enumerator-0.4.20...
Building hashable-1.2.3.3...
Building blaze-builder-0.4.0.1...
Building enumerator-0.4.20...
Installed hashable-1.2.3.3
Configuring case-insensitive-1.2.0.4...
Building case-insensitive-1.2.0.4...
Installed blaze-builder-0.4.0.1
Configuring scientific-0.3.3.8...
Building scientific-0.3.3.8...
Configuring unordered-containers-0.2.5.1...
Installed case-insensitive-1.2.0.4
Building unordered-containers-0.2.5.1...
Configuring streaming-commons-0.1.13...
Installed scientific-0.3.3.8
Installed enumerator-0.4.20
Configuring attoparsec-0.13.0.1...
Building streaming-commons-0.1.13...
Building attoparsec-0.13.0.1...
Configuring zlib-enum-0.2.3.1...
Installed vector-0.11.0.0
Building zlib-enum-0.2.3.1...
Installed unordered-containers-0.2.5.1
Installed zlib-enum-0.2.3.1
Installed streaming-commons-0.1.13
Configuring blaze-builder-enumerator-0.2.1.0...
Building blaze-builder-enumerator-0.2.1.0...
Installed blaze-builder-enumerator-0.2.1.0
Installed attoparsec-0.13.0.1
Configuring attoparsec-enumerator-0.3.4...
Building attoparsec-enumerator-0.3.4...
Installed attoparsec-enumerator-0.3.4
Failed to install snap-core-0.9.8.0
Build log ( /Users/trevorrothaus/Dropbox/Documents/www/elm/test/.cabal-sandbox/logs/snap-core-0.9.8.0.log ):
cabal:
/Users/trevorrothaus/Dropbox/Documents/www/elm/test/.cabal-sandbox/logs/snap-core-0.9.8.0.log:
does not exist
```

**Comment by jvoigtlaender** [comment]
Ah, okay. By "running again", I meant repeating the whole exercise of the README. But I didn't say that explicitly indeed.

About the issue at hand: Hmm. So nothing to do with Elm at all. I have no idea what lets the `snap-core-0.9.8.0` install fail. Maybe try to get help from the Snap people at http://snapframework.com/? Their issue tracker is at https://github.com/snapframework/snap-core/issues.

**Comment by jvoigtlaender** [comment]
Just for the record: Installing snap-core-0.9.8.0 works without problem on my machine. So it's not a general issue of Snap. There must be something about your version of GHC or your version of Cabal or your system environment that leads to this breakdown. What versions of GHC and Cabal are you at?

**Comment by jvoigtlaender** [comment]
Another thought. According to the dry-run, the dependency that would have been installed next in the sequence would have been unix-compat-0.4.1.4. Possibly that one is the culprit. Try installing that one in a fresh sandbox?

**Comment by trotha01** [comment]
```
$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 7.10.1
```
```
$ cabal --version
cabal-install version 1.22.2.0
using version 1.22.2.0 of the Cabal library
```

and
```
$ cabal install unix-compat-0.4.1.4
Resolving dependencies...
All the requested packages are already installed:
unix-compat-0.4.1.4
Use --reinstall if you want to reinstall anyway.
```

I actually went through attoparsec-enumerator-0.3.4 and down, but got the same "already installed" message for all of them.

Then trying ``` snap-core-0.9.8.0 ``` again gives the same ```.cabal-sandbox/logs/snap-core-0.9.8.0.log:
does not exist``` error

**Comment by jvoigtlaender** [comment]
Sorry, I'm out of my wisdom, then. I have GHC 7.10.2, and can install that package. Really seems like an issue for the Snap people now. (Maybe it's already known that Snap can't be built with GHC 7.10.1.)

**Comment by trotha01** [comment]
I appreciate the help, I'll keep looking into this. I'll close the issue, since it is not related to elm-lang.org.

I can add a comment if I find out the solution.

**Comment by trotha01** [comment]
I found this article:
http://www.yesodweb.com/blog/2015/06/cabals-does-not-exist-error-message

The fix was to use the fpcomplete mirror of hackage
https://www.fpcomplete.com/blog/2015/03/hackage-mirror

tl;dr
replace the remote-repo line in your ~/.cabal/config with the following:
```remote-repo: hackage.fpcomplete.com:http://hackage.fpcomplete.com/```

then run ``` cabal update ```
